src/button-dropdown/internal.tsx

```diff
@@ -230,9 +230,7 @@ const InternalButtonDropdown = React.forwardRef(
           {text}
         </InternalButton>
       );
-      trigger = showMainActionOnly ? (
```

Comment:
This change is needed so that onKeyDown and onKeyUp handlers are attached that prevent Enter keypress propagation. Without that, the main action cannot be activated with Enter.

Reply:
I am not 100% sure, but I think that is because those should work for both the trigger and the dropdown: if the listeners are added to the button dropdown trigger alone, that might change the current behaviour. I can experiment with that but in a standalone PR.

Reply:
Is this fixing an existing bug? If so, can we add coverage for the fix? I'd be OK with having this in a separate PR (actually even better).

Reply:
Created a separate PR: https://github.com/cloudscape-design/components/pull/2897